src/jdk.jfr/share/classes/jdk/jfr/internal/Utils.java

```diff
@@ -719,6 +725,20 @@ public static String formatDuration(Duration d) {
         }
     }
 
+    public static boolean shouldSkipBytecode(String eventName, Class<?> superClass) {
+        if (!superClass.getName().equals("jdk.jfr.events.AbstractJDKEvent")) {
+            return false;
+        }
+        return eventName.startsWith("jdk.Container") && getMetrics() == null;
+    }
+
+    private static Metrics getMetrics() {
+        if (metrics == null) {
+            metrics = Metrics.systemMetrics();
```

Comment:
Will this not lead to a lookup every time in an non-container environment?

Reply:
Yes. Now I see why you used `Metrics[]` - will fix.